source/fab/steps/grab.py

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+import os
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Dict
+
+from fab.steps import Step
+from fab.util import run_command
+
+
+class GrabBase(Step, ABC):
+    """
+    All grab steps require a source and a folder in which to put it.
+
+    """
+    def __init__(self, src, dst_label, name=None):
+        """
+        Args:
+            - src: The source location to grab. The nature of this is depends on the subclass.
+            - dst_label: The name of a sub folder in the project workspace, in which to put the source.
+
+        """
+        super().__init__(name=name or f'{self.__class__.__name__} {dst_label}'.strip())
+        self.src: str = src
+        self.dst_label: str = dst_label
+
+    @abstractmethod
+    def run(self, artefacts: Dict, config):
+        super().run(artefacts, config)
+        if not config.source_root.exists():
+            config.source_root.mkdir(parents=True, exist_ok=True)
```

Comment:
It should really be the responsibility of the config object to create it's source folder.